keps/sig-network/3698-multi-network/README.md

```diff
@@ -0,0 +1,1114 @@
+18. Kubelet network-based probing can be optionally enabled for the non- Cluster
+    Default PodNetwork attachments to Pod.
+19. Kubernetes API can be optionally accessible via the non- Cluster Default
+    PodNetwork attachments to Pod.
+20. Kubernetes API can optionally reach the Pod via the non- Cluster Default
+    PodNetwork attachments.
+21. All PodNetwork attachments to Pod are optionally able to provide Service
+and NetworkPolicy functionality.
+
+### Risks and Mitigations
+
+N/A
+
+## Background
+
+### Network
+Network is a very overloaded term, and for many might have different meaning: it
+might be represented as a VLAN, or a specific interface in a Node, or identified
+as a unique IP subnet (a CIDR). In this design we do not want to limit to one
+definition, and ensure that we are flexible enough to satisfy anyone's definition.
+
+### CNI usage models
+For the purpose of this design we want to call out the 2 models on how CNI API
+is used by various implementations.
+
+#### Standalone
+This is the simplest and direct mode. Here we leverage in full the CNI API and
+all its configuration capabilities. All parameters required to set up a network
+inside a Pod network namespace are provided from the default kubelet API and the
+parameters of the local conflist file present in CNI configuration path (default
+/etc/cni/net.d/). The binary configures everything based on just these parameters
+and does not require any additional connection with Kubernetes API.
+
+<p align="center">
+  <img src="cni-stand.png?raw=true" alt="standalone CNI mode"/>
+</p>
+
+#### Agent-based
+In this model, the CNI implementation is based on a fully fledged agent that
+usually runs as a hostNetwork Pod on each of the cluster Nodes. That agent has
+access to the Kubernetes API with RBAC defined permissions. The CNI binary is as
+simple as possible and communicates with the agent via local connections (e.g.
+socket file). Most of the logic is inside the agent. This agent does not rely on
+the CNI conflist, but gathers all the required data from Kubernetes API on its
+own (e.g. pulls the whole Pod object).
+
+<p align="center">
+  <img src="cni-agent.png?raw=true" alt="agent-based CNI mode"/>
+</p>
+
+## Phase I Design Details
+### Overview
+This design adds information to Pods to which “networks” (plural) it attaches to.
+Then the Pod scheduler will be able to understand which “network” is available
+and which is not in the cluster.
+
+For that purpose this design introduces 2 new objects: *PodNetwork* and
+*PodNetworkAttachment*. *PodNetwork* is a core component that functions as a
+representation of a specific network in Kubernetes cluster, as well as a future
+handle for other objects to reference.
+The *PodNetworkAttachment* is used as an additional Pod-level means to parameterize
+Pods’ network attachments.
+
+Lastly we wish to introduce a new controller for these objects, to support the
+proposed life cycle.
```

Comment:
I think you got the most of it. We do not propose CRI API change since Pod is an argument for these APIs. The container runtimes have to update to understand the new Pod fields.